app/assets/javascripts/components/utils.js.coffee

```diff
@@ -99,18 +99,32 @@ class @Utils
         $('body').css(cursor: 'auto')
       .done (json) =>
         Utils.showDynamicModal """
-          <h5>Log</h5>
-          <pre class="agent-dry-run-log"></pre>
-          <h5>Events</h5>
-          <pre class="agent-dry-run-events"></pre>
-          <h5>Memory</h5>
-          <pre class="agent-dry-run-memory"></pre>
+          <!-- Nav tabs -->
+          <ul id="resultTabs" class="nav nav-tabs agent-dry-run-tabs" role="tablist">
+            <li role="presentation"><a href="#tabEvents" aria-controls="tabEvents" role="tab" data-toggle="tab">Events</a></li>
+            <li role="presentation"><a href="#tabLog" aria-controls="tabLog" role="tab" data-toggle="tab">Log</a></li>
+            <li role="presentation"><a href="#tabMemory" aria-controls="tabMemory" role="tab" data-toggle="tab">Memory</a></li>
+          </ul>
+          <!-- Tab panes -->
+          <div class="tab-content">
+            <div role="tabpanel" class="tab-pane" id="tabEvents">
+              <pre class="agent-dry-run-events"></pre>
+            </div>
+            <div role="tabpanel" class="tab-pane" id="tabLog">
+              <pre><small class="agent-dry-run-log"></small></pre>
+            </div>
+            <div role="tabpanel" class="tab-pane" id="tabMemory">
+              <pre class="agent-dry-run-memory"></pre>
+            </div>
+          </div>
           """,
           body: (body) ->
             $(body).
               find('.agent-dry-run-log').text(json.log).end().
               find('.agent-dry-run-events').text(json.events).end().
               find('.agent-dry-run-memory').text(json.memory)
+            active = if json.events.match(/^\[?\s*\]?$/) then 'tabLog' else 'tabEvents'
```

Comment:
Defines the tab will be activated according to checking whether the events are empty. If empty, the Logs tab is activated.